### PR TITLE
Implemented UserDB, AbstractUserDB, their methods, and refactored the User class

### PR DESCRIPTION
**Pull request by dylancoffey99**

### CODE REVIEW

**Review by nadinefrawley — COMMENTED** [review]
This all looks good and well written! Good job implementing the type casting to the user class methods!

**Review by nadinefrawley — APPROVED** [review]
This looks good and well written! Good job implementing type casting to the user class methods!

**Comment by wasimghazal** [comment]
it's looking good and I think we can pull it, just try to call the user file from the abstract_db_handler but it give me error I think it will be easy to fix this.

File: system/models/database/abstract_db_handler.py [hunk 1]
```diff
@@ -1,5 +1,6 @@
 from abc import ABC, abstractmethod
 from typing import List
+from system.models.users.user import User
```
Comment (wasimghazal) [inline system/models/database/abstract_db_handler.py:3]: I get an error here when tried to call the user file, but I think that OK we can fix that later

File: system/models/users/user.py [hunk 2]
```diff
@@ -9,21 +10,21 @@ def __init__(self, user_id: int, user_name: str, password: str, is_admin: bool):
...
+    def get_password(self) -> str:
         return self._password
 
-    def get_is_admin(self):
+    def get_is_admin(self) -> bool:
         return self._is_admin
 
+    def get_user_as_list(self) -> List[str]:
+        return [str(self._user_id), self._user_name, self._password, str(self._is_admin)]
+
```
Comment (wasimghazal) [inline system/models/users/user.py:27]: The file looks good, just this function "get_user_as_list" I didn't understand why we need to return a list of all entry

**Review by wasimghazal — APPROVED** [review]

**Pull request merged**